devito/arguments.py

```diff
@@ -77,14 +77,31 @@ class ScalarArgument(Argument):
     def __init__(self, name, provider, reducer=lambda old, new: new, default_value=None):
         super(ScalarArgument, self).__init__(name, provider, default_value)
         self.reducer = reducer
+        self._frozen = False
+
+    def reset(self):
+        super(ScalarArgument, self).reset()
+        self._frozen = False
 
     def verify(self, value, enforce=False):
-        # Assuming self._value was initialised as appropriate for the reducer
+        # If we're being passed a null value, ignore
         if value is not None:
             if self._value is not None and not enforce:
-                self._value = self.reducer(self._value, value)
+                # We already have a value and the value passed
+                # here is not enforced (i.e. optional)
```

Comment:
can you remind me the use case for enforced?

Reply:
Added docstring to `verify` for explanation